Event page and share button

litviakk opened the pull request:
Added a very drafty event page with a share button for testing 
<img width="1292" alt="Screen Shot 2024-09-13 at 11 35 04 AM" src="https://github.com/user-attachments/assets/ec8d5be7-bce1-48ea-a0e3-60319056f174">
What I don't like:
* Event ID in URL, a URL Slug based on the event name might look better
* The button is ugly
* Content of the share needs to be thought through

Also, it seems there is an option of sharing a file, so it's possible to share the ICAL file, though not really sure how useful it is

Let me know your thoughts about it

gregsadetsky commented:
great work! I'll get back to you with some thoughts and we should definitely look in @nycnoise as well! Jessica, I can/will explain over a call

litviakk commented:
Testing on a phone failed...
![IMAGE 2024-09-13 12:32:06](https://github.com/user-attachments/assets/ef6dfc4a-9fc2-4563-ab0f-709bd01254ed)
Likely bc it's not [https](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share)
If you have a dev server w/ https handy to test it out it would be nice

p.s. the button is looking much better there

gregsadetsky commented:
you can use ngrok to get a local https proxy

litviakk commented:
nice tool! indeed it was https, works like a charm
![IMAGE 2024-09-13 15:36:51](https://github.com/user-attachments/assets/dddf1996-479d-4940-8518-2b610ca6c0f5)

gregsadetsky commented:
Super! I have a few thoughts about this feature - maybe it'd be easier to chat about it (I just emailed you)

Quick q - we can activate the share from anywhere right? It doesn't have to be a button - it's just a javascript call right?

My feeling right now is 1) let's add the share to the events on the homepage (and month pages) - probably a share icon 2) the shared url could have the event id - it's not as elegant as a slug but it works 3) I think it might make sense to not show a page that has only the event, but rather show the full month with the event highlighted and scrolled to

So /?event=123 would be the equivalent of /2024-09/ but scrolled down to the event + the event highlighted

Does that make sense?

litviakk commented:
it does! and it's much nicer and simpler than having a page for an event!
and yes it's a basic js event listener, so we can call it whenever

litviakk commented:
Added a sharable icon to the index page + anchors to link to + some ugly highlighting

Couple of Qs:
What's the proper way to rebake HTML for events? Just calling a `rebake_html` function on the events doesn't seem to work for some reason

Are the monthly pages static? What would be a way to render them? Just delete from the db and let it refresh?

gregsadetsky commented:
just tested it, some small notes:
- we'd want cursor:pointer to get a cursor pointer on desktop
- on mobile (via ngrok), it works, but when selecting "copy" as the action (or email), the copied result is not just the url, it also has "some blurb" which we wouldn't want. we'd only want url.
- as we talked, the event url should always include the month name
- could you check if django does receive the full full url including the hashtag i.e. `#event-...` part? if so, we'll be able to set the correct data as the opengraph tags i.e. we want it so that when people paste `nyc-noise.com/../#event...` in an imessage, django returns the correct opengraph value for that url... we'd also want to make sure that imessage/opengraph in general sees that a `nyc-noise.com/.../#event-123` and `nyc-noise.com/.../#event-456` are actually different urls and have different opengraph previews

let me know if you have any questions!

also, to rebake the events, you can run:

`python manage.py resave_all_events`

you should see the share button appear then

litviakk commented:
so... turns out that #achors are not[ a part of URL](https://stackoverflow.com/a/3081862) and not being passed by browsers as part of HTTP request...
I guess if we want to have different og tags we have to have different URLs
other than that I fixed everything lol
there is a question though - what if an event has an overwritten date and happens to be moved to a different month (e.g. from 10/31 to 11/1)? basically, should we consider an overwritten date when we generate the link?

gregsadetsky commented:
Ahh, I thought it might happen! That sucks but isn't a huge problem

So ... I think we'll need to make event pages again with distinct urls. Let's have /event/id and the page itself can be empty and just have the right OG tags, AND it should use a <meta to redirect the user to /(month)/#eventid to get the highlight behavior.

Does this make sense?

This also fixes the issue with events that are moved to another month (which is a good point!)

litviakk commented:
It does!
However, I would like to return to the discussion about using slugs instead
of event ids.
I believe we would need to use dates if we want to have unique URLs for the
events, and we could keep track of the date overwrites to have the old
links (which might have been shared before the reschedule) working too and
redirect them to the month page.
Anyway, if you want to go live without them we can add them later, just
need to keep the event/{id} URLs working on the back

k

On Tue, Oct 15, 2024 at 5:44 PM Greg Sadetsky ***@***.***>
wrote:

> Ahh, I thought it might happen! That sucks but isn't a huge problem
>
> So ... I think we'll need to make event pages again with distinct urls.
> Let's have /event/id and the page itself can be empty and just have the
> right OG tags, AND it should use a <meta to redirect the user to
> /(month)/#eventid to get the highlight behavior.
>
> Does this make sense?
>
> This also fixes the issue with events that are moved to another month
> (which is a good point!)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gregsadetsky/nycnoise/pull/318#issuecomment-2415219913>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADC4I4AIXNAURVPVL7NMTLTZ3WECXAVCNFSM6AAAAABOFUAOR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJVGIYTSOJRGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

gregsadetsky commented:
> It does! However, I would like to return to the discussion about using slugs instead of event ids. I believe we would need to use dates if we want to have unique URLs for the events, and we could keep track of the date overwrites to have the old links (which might have been shared before the reschedule) working too and redirect them to the month page. Anyway, if you want to go live without them we can add them later, just need to keep the event/{id} URLs working on the back k

I'd go with slugs usually (they look nicer, they might have some seo impact, and they don't facilitate iterating through all events) but in this case, as these urls are only there for the opengraph sharing, keeping them short and stable (the ID is obviously stable) seem pretty advantageous.

I also like the old schoolness of just /(some number) :-)

so yeah maybe let's start with /id and if we realize it was a mistake, we can pivot later.

thanks!

litviakk commented:
* Added an empty page for the events
* Changed the icon and moved it under 'TIX' link
* Clean up the share info (only the link now)
* Clean up styles

Couple of concerns - 
1 - for some reason highlighting has broken, trying to figure out why...
2 - when you open an event link - the event date header is located right on the opened event, it's especially pitiful on mobile where you can't really see the shared event under the date, playing w/ scroll-margin-top to make it better

litviakk commented:
See that we have dates in UTC... will fix momentarily

litviakk commented:
Okay, here we go:
<img width="429" alt="Screen Shot 2025-01-20 at 11 37 09 AM" src="https://github.com/user-attachments/assets/731990c2-2369-480c-adfe-e8b6b04365c4" />

gregsadetsky commented:
Great work! Will review

Just fyi you can use https://www.opengraph.xyz to check the opengraph tags (but yeah it's also always a good idea to check on specific social sites/apps)

litviakk commented:
yep, I tried it, but it doesn't get past ngrok...
<img width="427" alt="Screen Shot 2025-01-20 at 3 54 38 PM" src="https://github.com/user-attachments/assets/831a39c1-fc28-47be-8b26-27b830d7d0cf" />

gregsadetsky commented:
MERGED!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

thank you so so much

rolling out to prod now, I'll babysit it
